### PR TITLE
New Improved Reform api docs

### PR DESCRIPTION
**Pull request by alectronic0**
screenshot from 2018-07-02 20-02-09

Over the last couple of weeks I've been crossing referencing our environments and github to get a good idea of repos.

I have then replaced the previous microservice details for this.
and re added dependency links.

What I still need to do:
- [ ]  Talk with services to see what they depend on with in CC.
- [x]  Pick better Colours
- [ ] Write up a doc to automate uploading to there repos swagger docs.
- [ ] maybe added reference of libraries that use said CC or servicer
- [ ] Maybe show connections to 3rd party Gov.uk services (if we use them)
- [ ] Better outline teams domains

### CODE REVIEW

File: docs/microservices.json [hunk 1]
```diff
@@ -1,115 +1,300 @@
...
     "apis": [
+
         {
-            "id": "draft-store",
-            "name": "Draft Store",
-            "spec": "https://hmcts.github.io/reform-api-docs/specs/draft-store.json",
-            "group": "Platform Engineering",
-            "repository": "https://github.com/hmcts/draft-store",
+            "id": "idam-web",
+            "name": "IDAM (Tatical)",
```
Comment (doncem) [inline docs/microservices.json]: Missing `c` here in `Tactical`

File: docs/microservices.json [hunk 2]
```diff
@@ -246,42 +454,123 @@
...
+                    "hard": true,
+                    "apis": []
+                }
+            ],
+            "apis": [],
+            "version": null
+        },
+        {
+            "id": "em-anno",
+            "name": "EM Anno",
```
Comment (doncem) [inline docs/microservices.json]: Shall we just name fully: `EM Annotation`

File: docs/microservices.json [hunk 3]
```diff
@@ -307,25 +597,469 @@
...
+            "apis": [],
+            "version": null
+        },
+        {
+            "id": "rpe-feature-toggle-web",
+            "name": "RPE Feature Toggle Web",
+            "group": "RPE",
+            "proType": "web",
+            "description": "",
+            "repository": "https://github.com/hmcts/ft-toggle-web",
```
Comment (doncem) [inline docs/microservices.json]: this project was a spike and had to be within hmcts to test out pipeline. Should be removed

File: docs/microservices.json [hunk 4]
```diff
@@ -307,25 +597,469 @@
...
+            "proType": "api",
+            "description": "",
+            "repository": "https://github.com/hmcts/draft-store",
+            "spec": "https://hmcts.github.io/reform-api-docs/specs/draft-store.json",
+            "dependencies": [
+                {
+                    "id": "rpe-service-auth-provider",
+                    "hard": true,
+                    "apis": []
+                }
```
Comment (doncem) [inline docs/microservices.json]: I think draft store also depends on idam as it checks user details by making a call to it

File: docs/microservices.json (hunk 1 above)
Comment (alectronic0) [reply docs/microservices.json]: done

File: docs/microservices.json (hunk 4 above)
Comment (alectronic0) [reply docs/microservices.json]: added

**Pull request merged**